subgraphs/aave-forks/protocols/aave-v3/src/mapping.ts

```diff
@@ -0,0 +1,832 @@
+  if (!marketId || !Market.load(marketId.toHexString())) {
+    log.warning("[handleReserveDataUpdated]Market not found for reserve {}", [
+      event.params.reserve.toHexString(),
+    ]);
+    return;
+  }
+
+  const market = Market.load(marketId.toHexString())!;
+  const assetPriceUSD = getAssetPriceInUSDC(
+    Address.fromString(market.inputToken),
+    Address.fromString(protocol.priceOracle),
+    event.block.number
+  );
+
+  // TODO: better to pass market and protocol
+  _handleReserveDataUpdated(
+    event,
+    event.params.liquidityRate,
+    event.params.liquidityIndex,
+    event.params.variableBorrowRate,
+    event.params.stableBorrowRate,
+    protocolData,
+    marketId,
+    assetPriceUSD
+  );
+}
+
+export function handleReserveUsedAsCollateralEnabled(
+  event: ReserveUsedAsCollateralEnabled
+): void {
+  const marketId = getMarketIdFromToken(event.params.reserve);
+  if (!marketId) {
+    log.error(
+      "[handleReserveUsedAsCollateralEnabled]Failed to find market for asset {}",
+      [event.params.reserve.toHexString()]
+    );
+    return;
+  }
+  // This Event handler enables a reserve/market to be used as collateral
+  _handleReserveUsedAsCollateralEnabled(
+    marketId,
+    event.params.user,
+    protocolData
+  );
+}
+
+export function handleReserveUsedAsCollateralDisabled(
+  event: ReserveUsedAsCollateralDisabled
+): void {
+  const marketId = getMarketIdFromToken(event.params.reserve);
+  if (!marketId) {
+    log.error(
+      "[handleReserveUsedAsCollateralDisabled]Failed to find market for asset {}",
+      [event.params.reserve.toHexString()]
+    );
+    return;
+  }
+  // This Event handler disables a reserve/market being used as collateral
+  _handleReserveUsedAsCollateralDisabled(
+    marketId,
+    event.params.user,
+    protocolData
+  );
+}
+
+export function handleDeposit(event: Supply): void {
+  const marketId = getMarketIdFromToken(event.params.reserve);
+  if (!marketId) {
+    log.error("[handleDeposit]Failed to find market for asset {}", [
+      event.params.reserve.toHexString(),
+    ]);
+    return;
+  }
+  _handleDeposit(
+    event,
+    event.params.amount,
+    marketId,
+    protocolData,
+    event.params.onBehalfOf
+  );
+}
+
+export function handleWithdraw(event: Withdraw): void {
+  const marketId = getMarketIdFromToken(event.params.reserve);
+  if (!marketId) {
+    log.error("[handleWithdraw]Failed to find market for asset {}", [
+      event.params.reserve.toHexString(),
+    ]);
+    return;
+  }
+  _handleWithdraw(
+    event,
+    event.params.amount,
+    marketId,
+    protocolData,
+    event.params.user
+  );
+}
+
+export function handleBorrow(event: Borrow): void {
+  const marketId = getMarketIdFromToken(event.params.reserve);
+  if (!marketId) {
+    log.error("[handleBorrow]Failed to find market for asset {}", [
+      event.params.reserve.toHexString(),
+    ]);
+    return;
+  }
+
+  // Set reserveFactor if not set, as setReserveFactor may be never called
```

Comment:
nice job

Reply:
Why choose to do it in Borrows? I think it would be cleaner to make this a helper function.

Reply:
Good point. Moved the code to a separate function